From 458567b4009816878b1a5e67b85c6263e7a5aa62 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Tue, 26 Feb 2019 19:57:49 +0300
Subject: [PATCH 03/15] hotplug: Remove '/dev/' prefix from DEVNAME variable

This can be happened when hotplug works together with udev (eudev).
In this case, nested folder 'dev' is created in the root folder '/dev'.

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 plug/hotplug.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/plug/hotplug.c b/plug/hotplug.c
index de0511f..f6ef318 100644
--- a/plug/hotplug.c
+++ b/plug/hotplug.c
@@ -594,7 +594,10 @@ static void hotplug_handler(struct uloop_fd *u, unsigned int ev)
 
 		if (e) {
 			*e = '\0';
-			blobmsg_add_string(&b, &buf[i], &e[1]);
+			if ((strcmp(&buf[i], "DEVNAME") == 0) && (strncmp(&e[1], "/dev/", 5) == 0))
+				blobmsg_add_string(&b, &buf[i], &e[6]);
+			else
+				blobmsg_add_string(&b, &buf[i], &e[1]);
 		}
 		i += l;
 	}
-- 
2.28.0.windows.1

